keps/sig-network/20190125-ingress-api-group.md

```diff
@@ -161,146 +159,157 @@ consistent with the syntax supported by any of the common proxy vendors:
+
+> `Host` can be "precise" which is an fully-qualified domain name
+> without the terminating dot of a network host (e.g. "foo.bar.com")
+> or "wildcard", which is a domain name prefixed with a single
+> wildcard label (e.g. "*.foo.com").
+>
+> Requests will be matched against the `Host` field in the following
+> way:
+>
+> If `Host` is precise, the request matches this rule if the http host
+> header is equal to `Host`.
+>
+> If `Host` is a wildcard, then the request matches this rule if the
+> http host header is to equal to the suffix (removing the first
+> label) of the wildcard rule. The wildcard character '*' must appear
+> by itself, be the first label and matches only a single label.
+>
+> Examples:
+>
+> - "*.foo.com" matches "bar.foo.com" because they share an equal
+>   suffix "foo.com".
+> - "*.foo.com" does not match "aaa.bbb.foo.com" as the wildcard only
+>   matches a single label.
+> - "*.foo.com" does not match "foo.com", as the wildcard must match a
+>   single label.
+>
+> Note: label refers to a "DNS label", e.g. the strings separated by
+> the dots "." in the domain name.
 
 ### Healthchecks
 
-The current spec does not have any provisions to customize appropriate
-healtchecks for referenced backends. Many users already have a healthcheck
-URLthat is lightweight and different from the HTTP root (e.g. `/`).
-
-One option that has been explored is to infer the healthcheck URL from the
-Readiness probes on the Pods of the Service. This method has proven to be quite
-unstable: Every Pod in a Service can have a different Readiness probe definition
-and it's not clear which one should be used. Furthermore, the behavior is quite
-implicit and creates action-at-a-distance relationship between the Ingress and
-Pod resources.
-
-#### Proposal
+The current spec does not have any provisions to customize
+healthchecks for referenced backends. Many users already have a
+healthcheck URL that is lightweight and different from the HTTP root
+(i.e. `/`).
+
+One obvious question that arises is why the Ingress healthcheck
+configuration is (a) is needed and (b) is different from the current
+Pod readiness and liveness checks. The Ingress healthcheck represents
+an end-to-end check from the proxy server to the backend.  The
+Kubelet-based service health check operates only within the VM and
+does not include the network path. A minor point is that it is also
+the case that some providers require a healthcheck to be specficied as
+part of load balancing.
+
+An option that has been explored is to infer the healthcheck URL from
+the Readiness/Liveness probes on the Pods of the Service. This method
+has proven to be unworkable: Every Pod in a Service can have a
+different Readiness probe definition and therefore it's not clear
+which one should be used. Furthermore, the behavior is implicit and
+creates action-at-a-distance relationship between the Ingress and Pod
+resources.
+
+#### Healthchecks proposal
```

Comment:
This should not be done at all.
What does a health check mean for an ingress object?
A better way would be to configure it on the service and write this to the endpoints.
It's also defined always on the member pool in any load balancer, that has active health checks.

Why we should not have this at all in ingress: 
The contract is the endpoints list will be updated as soon endpoint is ready/not ready. There is no active health checking needed in this case.

Reply:
See comments below about healthcheck.